internal/oscal/generate_test.go

```diff
@@ -29,3 +60,171 @@ func TestFmtFieldName(t *testing.T) {
 		}
 	}
 }
+
+// TestGenerateUniqueIdMapId tests that the generateUniqueIdMap function
+// returns the correct '$id'.
+func TestGenerateUniqueIdMapId(t *testing.T) {
+	oscalMap, err := parseOscalFileToMap()
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, actualId := generateUniqueIdMap(oscalMap)
+
+	expectedId := "#assembly_oscal-component-definition_component-definition"
```

Comment:
[no change required] I wonder if this id should be a `const` somewhere? yes it would be hardcoded - but maybe more accurate than the title check we perform in this function we're testing. food for thought.

Reply:
Updated it to be `const expectedId string = "#assembly_oscal-component-definition_component-definition"`

I'm thinking we maybe shouldn't make that a global constant that is reused in other tests though. I would think we want to use the actual return value id from calling the actual function in the tests.

Reply:
Thoughts?

Reply:
ah yes - I should have clarified - I meant that this may make a good candidate for a global constant that would be used in the `generateUniqueIdMap` function in place of the current title comparison that we use currently. 

Then it could be used in this test as well.

Reply:
I think I'm tracking with you now. Updated it to a global constant, and removed the check/discovery of it in the `generateUniqueIdMap` function.